ciao-controller/compute.go

```diff
@@ -453,11 +453,12 @@ func instanceToServer(context *controller, instance *types.Instance) (payloads.S
 }
 
 // returnErrorCode returns error codes for the http call
-func returnErrorCode(w http.ResponseWriter, httpError int, message string) {
+func returnErrorCode(w http.ResponseWriter, httpError int, message ...string) {
 	var returnCode payloads.HTTPReturnErrorCode
 	returnCode.Error.Code = httpError
 	returnCode.Error.Name = http.StatusText(returnCode.Error.Code)
-	returnCode.Error.Message = message
+
+	returnCode.Error.Message = strings.Join(message, " ")
```

Comment:
Sorry if my initial comment was confusing, but this is what I had in mind:

```
diff --git a/ciao-controller/compute.go b/ciao-controller/compute.go
index 75f5153..2ed9ce5 100644
--- a/ciao-controller/compute.go
+++ b/ciao-controller/compute.go
@@ -453,11 +453,11 @@ func instanceToServer(context *controller, instance *types.Instance) (payloads.S
 }
 
 // returnErrorCode returns error codes for the http call
-func returnErrorCode(w http.ResponseWriter, httpError int, message string) {
+func returnErrorCode(w http.ResponseWriter, httpError int, format string, args ...interface{}) {
        var returnCode payloads.HTTPReturnErrorCode
        returnCode.Error.Code = httpError
        returnCode.Error.Name = http.StatusText(returnCode.Error.Code)
-       returnCode.Error.Message = message
+       returnCode.Error.Message = fmt.Sprintf(format, args)
 
        b, err := json.Marshal(returnCode)
        if err != nil {
```

And then you can write code like:
```
returnErrorCode(w, http.StatusNotFound, "Instance %d could not be found", instanceID)
```